### DISCUSSION TITLE
How to add into Search path in Free Pascal?

### DISCUSSION THREAD

**Discussion by abitrolly** [Cooler2/ApusGameEngine#6] [Q&A]
In https://docs.google.com/document/d/1sl9x3FB-qI7e8DnW6dpUHevZSU8ddfsNHwwTk5ygYUs/edit there is this point

> 3. Go to the project options and add these folders into the Search path. Add also these conditional defines: 

I have a 0 knowledge of Free Pascal and Lazarus. I had to ask Google to discover that I need to exec `startlazarus` to run it, but it doesn't open `.dpr` files, so I am trying to use `fpc`. Unfortunately, it doesn't seem to find the lib.

```
$ fpc -h | grep earch
      -FD<x>     Set the directory where to search for compiler utilities
      -FM<x>     Set the directory where to search for unicode binary files
      -Xd        Do not search default library path (sometimes required for cross-compiling when not using -XR)
      -Xp<x>     First search for the compiler binary in the directory <x>
      -XR<x>     Prepend <x> to all linker search paths (BeOS, Darwin, FreeBSD, Linux, Mac OS, Solaris)
```
```
$ fpc NoGame.dpr -XRApusGameEngine
Free Pascal Compiler version 3.2.2 [2023/01/19] for x86_64
Copyright (c) 1993-2021 by Florian Klaempfl and others
Target OS: Linux for x86-64
Compiling NoGame.dpr
NoGame.dpr(2,6) Fatal: Can't find unit Apus.Engine.GameApp used by NoGame
Fatal: Compilation aborted
Error: /usr/bin/ppcx64 returned an error exitcode
```

**Comment by Cooler2** [comment]
I see. I made Lazarus project files (.lpi) for some demo projects, but not for all. For this case I'll add an explanation how to create a Lazarus project from Delphi project to the documentation, since now it's all about Delphi.

**Reply by abitrolly** [reply]
Is it possible to completely avoid Lazarus/Delphi GUI to create a game with the engine?

**Reply by Cooler2** [reply]
> Is it possible to completely avoid Lazarus/Delphi GUI to create a game with the engine?

Sure! Lazarus is just an IDE built over FPC, you can use "Project Options"->"Show Options" button to view the full command line which builds the project. That command can be run without Lazarus and will produce the same result.
